src/udev/scsi_id/scsi_serial.c

```diff
@@ -700,19 +867,26 @@ static int do_scsi_page83_prespc3_inquiry(struct scsi_id_device *dev_scsi, int f
 }
 
 /* Get unit serial number VPD page */
-static int do_scsi_page80_inquiry(struct scsi_id_device *dev_scsi, int fd,
-                                  char *serial, char *serial_short, int max_len) {
+static int sgio_do_scsi_page80_inquiry(struct scsi_id_device *dev_scsi, int fd,
+                                       char *serial, char *serial_short, int max_len) {
         int retval;
-        int ser_ind;
-        int i;
-        int len;
         unsigned char buf[SCSI_INQ_BUFF_LEN];
 
         memzero(buf, SCSI_INQ_BUFF_LEN);
-        retval = scsi_inquiry(dev_scsi, fd, 1, PAGE_80, buf, SCSI_INQ_BUFF_LEN);
+        retval = sgio_scsi_inquiry(dev_scsi, fd, 1, PAGE_80, buf, SCSI_INQ_BUFF_LEN);
         if (retval < 0)
                 return retval;
 
+        retval = process_page80_inquiry_data(dev_scsi, serial, serial_short, buf, max_len);
+        if (retval)
+                return 1;
+        return 0;
```

Comment:
Not necessary. Please propagate the result as is.

Reply:
Done